src/modules/client/pages/ni/courses/LearnersDirectory.vue

```diff
@@ -106,6 +141,67 @@ export default {
     getAvatar (link) {
       return link || DEFAULT_AVATAR;
     },
+    resetAddLearnerForm () {
+      this.firstStep = true;
+      this.addNewLearnerIdentityStep = false;
+      this.newLearner = { ...clear(this.newLearner) };
+      this.$v.newLearner.$reset();
+    },
+    async nextStepLearnerCreationModal () {
+      try {
+        this.$v.newLearner.$touch();
+        if (!this.newLearner.local.email || this.$v.newLearner.local.email.$error) {
+          return NotifyWarning('Champ(s) invalide(s).');
+        }
+
+        this.learnerCreationModalLoading = true;
+        const userInfo = await Users.exists({ email: this.newLearner.local.email });
+
+        if (userInfo.exists) {
+          if (!userInfo.user.company) {
+            await Users.updateById(userInfo.user._id, { company: this.company._id });
+            await this.addLearner();
+          } else if (userInfo.user.company === this.company._id) {
+            return NotifyWarning('L\'apprenant a déjà été ajouté.');
+          } else return NotifyNegative('L\'apprenant n\'est pas relié à cette structure.');
+        } else this.addNewLearnerIdentityStep = true;
+        this.firstStep = false;
+        this.$v.newLearner.$reset();
+      } catch (e) {
+        NotifyNegative('Erreur lors de l\'ajout de l\'apprenant.');
+      } finally {
+        this.learnerCreationModalLoading = false;
+      }
+    },
+    formatUserPayload () {
+      const payload = removeEmptyProps(this.newLearner);
+      if (get(payload, 'contact.phone')) payload.contact.phone = formatPhoneForPayload(this.newLearner.contact.phone);
+      return ({
+        ...payload,
+        company: this.company._id,
+      });
+    },
+    async addLearner () {
+      try {
+        this.learnerCreationModalLoading = true;
+        if (!this.firstStep) {
+          this.$v.newLearner.$touch();
+          if (!this.addnewLearnerIdentityStep && this.$v.newLearner.$invalid) return NotifyWarning('Champs invalides.');
```

Comment:
je ne suis pas sur de comprendre cette condition `!this.addnewLearnerIdentityStep` ?
et on n'utilise pas `$invalid` mais `$error` plutot

Reply:
en effet elle est fausse, il aurait fallut dire l'inverse mais comme on est deja dans `!this.firstStep` je peux l'enlever.